app/views/posts/index.html.erb

```diff
@@ -0,0 +1,33 @@
+<p id="notice"><%= notice %></p>
+
+<h1>Posts</h1>
+
+<table>
+  <thead>
+    <tr>
+      <th>Post</th>
+      <th colspan="3"></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @posts.each do |post| %>
+      <tr>
+        <td><%= post.post %></td>
+        
+        <td><%= link_to 'Show', post %></td>
+        <% if user_signed_in? %>
+        <td><%= post.user.username %></td>
+        <td><%= link_to 'Edit', edit_post_path(post) %></td>
+        <td><%= link_to 'Destroy', post, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <% end %>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+<% if !user_signed_in? %>
+  <%= link_to "Login", new_user_session_path, class:"button is-info" %>
+  <% end %>
+<br>
```

Comment:
* [ ] **_There should be NO application logic in the views (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...)._**
* [ ] **_View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb)_**
* [ ] **_There should be absolutely NO logic in views._**

Reply:
We don't know how to render an HTML file in a normal rb file. Our logic as far as we understand is not an application logic but is a view logic. Still, however, if it is not the right way, can you help us and give us some examples of how to do it?